Microsoft.Toolkit.Uwp.UI.Media/Geometry/Core/CanvasRoundRect.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Numerics;
+
+namespace Microsoft.Toolkit.Uwp.UI.Media.Geometry.Core
+{
+    /// <summary>
+    /// Structure which encapsulates the details of each of the core points
+    /// of the path of the rounded rectangle which is calculated based on
+    /// either the given (Size, CornerRadius, BorderThickness and Padding)
+    /// or (Size, RadiusX and RadiusY).
+    /// </summary>
+    internal struct CanvasRoundRect
```

Comment:
This struct is pretty massive (104 bytes). I'd recommend passing it by reference when used internally, and also changing those internal `Vector2` properties to fields, so that code accessing fields of those will be able to skip the temporary `Vector2` copy and to just index the target `float` field directly from the stack.

Reply:
Done

Reply:
Done. Removed all Vector2 properties and converted them to appropriate float properties.